### ISSUE TITLE
kms.Key: kms.Key.isLookupDummy incorrectly throws error when no-pre-existing-key / dummy-key is found

### ISSUE THREAD

**Issue by neoakris** [aws/aws-cdk#32368]
### Describe the bug

@go-to-k I just not around to testing the feature you added in https://github.com/aws/aws-cdk/issues/31574, and discovered: 
There was a problem in logic of original feature request (which has now been auto-closed.)  

The intent of the new feature was to allow:
1. check for existence of pre-existing kms key with given alias  
    ^--kms.Key.isLookupDummy() method was intended to do this, before the feature keyid would return a token which didn't allow an if statement check to occur, the new method returns true when key not found. 
    (**The bug is that it needlessly throws an error, which stops execution / doesn't make sense for an existence check. This is a relatively new feature and I suspect it never worked.**)
2. if (pre-existing kms with alias exists) use it
3. if (pre-existing kms with alias doesn't exist) create it then use it.

### Regression Issue

- [ ] Select this option if this issue appears to be a regression.

### Last Known Working CDK Version

Not Applicable

### Expected Behavior

Note: the value of this.config.kmsKeyAlias = alias/eks/dev1-eks (it'll show in error message)

```typescript
if (kms.Key.isLookupDummy(kms.Key.fromLookup(this.stack, "pre-existing-kms-key", { aliasName: this.config.kmsKeyAlias, returnDummyKeyOnMissing: true,  }))){
  console.log("pre-existing kms not found, will create, then use");
  eksBlueprint.resourceProvider(blueprints.GlobalResources.KmsKey, new blueprints.CreateKmsKeyProvider(
    this.config.kmsKeyAlias, {description: "Easy EKS generated kms key, used to encrypt etcd and ebs-csi-driver provisioned volumes"}
  ));
}
else {
  console.log("pre-existing kms key found, will use");
  eksBlueprint.resourceProvider(blueprints.GlobalResources.KmsKey, new blueprints.LookupKmsKeyProvider(this.config.kmsKeyAlias));
}
```

^--I expect this logic (described in laymans terms above) to work.

### Current Behavior

When I use `cdk deploy dev1-eks`  
to run the above logic snippet

It throws an error that stops everything, error shouldn't be thrown for an existance check.
```console
pre-existing kms not found, will create, then use
pre-existing kms not found, will create, then use
pre-existing kms not found, will create, then use
pre-existing kms not found, will create, then use
[Error at /dev1-eks] Could not find any key with alias named alias/eks/dev1-eks
Found errors
```

### Reproduction Steps

I'm going to ask if I can make some code public, which will make reproduction easier. (With luck that could happen around Jan/Feb)

### Possible Solution

Make the method not throw error when the isLookupDummy() is used

### Additional Information/Context

**Below is the workaround I'm currently using today:**
I'd like to replace it with a upstream cdk fix to feature of checking for existing kms, because the workaround has 3 negligible, yet undesirable side effects: 
* ensure's existance of aliased kms key prior to cdk's y/n prompt, so a key could be created even if you just want to dry run or cancel
* jq and aws become dependencies
* negligibly low risk of command injection.
<br/>

```typescript
function ensure_existance_of_aliased_kms_key(kmsKeyAlias: string){
    let kms_key:kms.Key;   
    const cmd = `aws kms list-aliases | jq '.Aliases[] | select(.AliasName == "${kmsKeyAlias}") | .TargetKeyId'`
    const cmd_results = execSync(cmd).toString();
    if(cmd_results===""){ //if alias not found, then make a kms key with the alias
        const create_key_cmd = `aws kms create-key --description="Generated kms key, used to encrypt etcd and ebs-csi-driver provisioned volumes"`
        const results = JSON.parse( execSync(create_key_cmd).toString() );
        const key_id = results.KeyMetadata.KeyId;
        const add_alias_cmd = `aws kms create-alias --alias-name ${kmsKeyAlias} --target-key-id ${key_id}`;
        execSync(add_alias_cmd);
    }
}

...(main logic)...
ensure_existance_of_aliased_kms_key(this.config.kmsKeyAlias)
eksBlueprint.resourceProvider(blueprints.GlobalResources.KmsKey, new blueprints.LookupKmsKeyProvider(this.config.kmsKeyAlias));
```

### CDK CLI Version

2.170.0

### Node.js Version

v20.18.1

### OS

Mac 15.1.1

### Language

TypeScript

### Language Version

5.4.5

**Comment by neoakris** [comment]
https://github.com/aws/aws-cdk/blob/v2.170.0/packages/aws-cdk-lib/aws-ssm/lib/parameter.ts#L587
^-- There's a variable named "ignoreErrorOnMissingContext"

so I think there's agreement in regards to my proposed solution to make the isLookupDummy() not throw an error, but the logic involved in that variable doesn't seem to prevent an error from being thrown.

**Comment by khushail** [comment]
Hi @neoakris , I tested the feature and here is my observation-

1. The key is returned when alias exists -

<img width="1128" alt="Screenshot 2024-12-11 at 2 07 58 PM" src="https://github.com/user-attachments/assets/e5d2d86f-a0f2-4b61-83fc-3f828ac5078c" />

2. Error is thrown during `cdk synth` when key does not exist -

<img width="684" alt="Screenshot 2024-12-11 at 2 14 20 PM" src="https://github.com/user-attachments/assets/684e69bd-9c99-48c5-ba80-ddad48c8b739" />


so as proposed in this https://github.com/aws/aws-cdk/pull/31676, no dummy value is returned in my case.

Are you pointing towards this scenario ?

**Comment by neoakris** [comment]
Hi khushail, thank you for taking some time to look into and test this.

**Summary:  
There's a good chance this is a [PEBCAK](https://en.wiktionary.org/wiki/PEBCAK) problem. If possible I'd like to wait until the next release of eks blueprints to confirm no integration issue exists (between cdk and eks-blueprints) before closing this, and that it's just because I messed up my packages.json**

-----------------

**Additional Details:**

I tried to create a short (relatively) self-contained example of the bug based on "aws-cdk": "2.170.0"
https://github.com/neoakris/aws-cdk_32368/blob/works_without_eks_blueprints/bin/cdk-main.ts  
but I noticed it worked (since you tested pre-exist and non-exist, I tested both cases and verified it worked.)

This lead me to believe it's some kind of integration issue with eks-blueprints, so I tried to integrate eks-blueprints into the example to reproduce the error.

Hum...  
per https://github.com/aws-quickstart/cdk-eks-blueprints/tree/blueprints-1.16.2?tab=readme-ov-file#getting-started-1  
**The current latest eks-blueprints is based on aws-cdk-lib@2.162.1 (b4 your fix came out), I wounder if that could be a problem. (regardless of any answers/responses) When they next release I'll re-test to see if it goes away.** 

When I first encountered the issue, I had managed to combine cdk-eks-blueprints 1.16.2, with a newer version of cdk 2.170.0, but things mostly seemed to work, but maybe this issue arose from me mixing cdk-eks-blueprint with a newer version of AWS CDK? instead of the version their project pins to.  

I forget the specifics of how I managed that so I'll replace the package.json with the one from the eks-blueprints project I was working on when I 1st found the issue to see if I can reproduce.

setting my package.json to be like the other, ended up with npm peer dependencies error.

I'm now woundering how this managed to work on my other setup, after clearing node_modules and trying to install fresh I noticed breakage there too. so this might be that I did something to my package.json file and it's getting stuck trying to use the old and new code at the same time, since the latest eks-blueprints uses an older version of cdk that predates your fix, and I somehow managed to do some tinkering to get it to temporarily partially work on my machine possibly in a wya with broken peer dependencies without realizing it. 

That probably means when I originally found the error it may have been due to a faulty local dev environment, that somehow allowed me to mix old and new version. So my bug report is likely invalid. Sorry for the noise and the trouble I've caused.
